raiden-network/light-client
# Convert epics to non-arrow functions and fix some network errors

Part of #1790 

**Short description**
So far, when an error failed with an error that should/could be handled inside the epic, the error logs on CLI were not very informative, and didn't contain the name of the function, because the epics functions were usually arrow functions. By using normal functions for the exported top level functions of the epics, and also including `epic.name` explicitly on the epics-specific error log, we get a better view of which epic failed.
This is just a stylistic change, and doesn't change anything in the epics logic.

An example of this error is the network error below, which made it explicit `monitorUdcBalanceEpic` was not properly handling failures there, and then a new commit fixed it.

While debugging it on CLI, it also was discovered that the force-exit after 10s was not getting triggered on self-shutdown of the SDK (only by the TERM/INT signal handlers), so this also got fixed. Hopefully, this won't be needed after we fix #2227 , so it's just a fallback right now.

```
2020-10-13T13:36:13.049Z [error] Epic error monitorUdcBalanceEpic [Function: monitorUdcBalanceEpic] Error: invalid response - 0
    at exports.XMLHttpRequest.request.onreadystatechange (/mnt/sec/Dev/brainbot/light-client/node_modules/.pnpm/ethers@4.0.48/node_modules/ethers/utils/web.js:84:29)
    at exports.XMLHttpRequest.dispatchEvent (/mnt/sec/Dev/brainbot/light-client/node_modules/.pnpm/ethers@4.0.48/node_modules/ethers/node_modules/xmlhttprequest/lib/XMLHttpRequest.js:591:25)
    at setState (/mnt/sec/Dev/brainbot/light-client/node_modules/.pnpm/ethers@4.0.48/node_modules/ethers/node_modules/xmlhttprequest/lib/XMLHttpRequest.js:610:14)
    at exports.XMLHttpRequest.handleError (/mnt/sec/Dev/brainbot/light-client/node_modules/.pnpm/ethers@4.0.48/node_modules/ethers/node_modules/xmlhttprequest/lib/XMLHttpRequest.js:532:5)
    at ClientRequest.errorHandler (/mnt/sec/Dev/brainbot/light-client/node_modules/.pnpm/ethers@4.0.48/node_modules/ethers/node_modules/xmlhttprequest/lib/XMLHttpRequest.js:459:14)
    at ClientRequest.emit (events.js:314:20)
    at Socket.socketErrorListener (_http_client.js:428:9)
    at Socket.emit (events.js:314:20)
    at emitErrorNT (internal/streams/destroy.js:92:8)
    at emitErrorAndCloseNT (internal/streams/destroy.js:60:3) {
  statusCode: 0,
  responseText: 'Error: getaddrinfo EAI_AGAIN parity.goerli.ethnodes.brainbot.com\n' +
    '    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:66:26)'
}
2020-10-13T13:36:13.061Z [info] %c action     color: #03A9F4; font-weight: bold {
  type: 'raiden/shutdown',
  payload: {
    reason: Error: invalid response - 0
        at exports.XMLHttpRequest.request.onreadystatechange (/mnt/sec/Dev/brainbot/light-client/node_modules/.pnpm/ethers@4.0.48/node_modules/ethers/utils/web.js:84:29)
        at exports.XMLHttpRequest.dispatchEvent (/mnt/sec/Dev/brainbot/light-client/node_modules/.pnpm/ethers@4.0.48/node_modules/ethers/node_modules/xmlhttprequest/lib/XMLHttpRequest.js:591:25)
        at setState (/mnt/sec/Dev/brainbot/light-client/node_modules/.pnpm/ethers@4.0.48/node_modules/ethers/node_modules/xmlhttprequest/lib/XMLHttpRequest.js:610:14)
        at exports.XMLHttpRequest.handleError (/mnt/sec/Dev/brainbot/light-client/node_modules/.pnpm/ethers@4.0.48/node_modules/ethers/node_modules/xmlhttprequest/lib/XMLHttpRequest.js:532:5)
        at ClientRequest.errorHandler (/mnt/sec/Dev/brainbot/light-client/node_modules/.pnpm/ethers@4.0.48/node_modules/ethers/node_modules/xmlhttprequest/lib/XMLHttpRequest.js:459:14)
        at ClientRequest.emit (events.js:314:20)
        at Socket.socketErrorListener (_http_client.js:428:9)
        at Socket.emit (events.js:314:20)
        at emitErrorNT (internal/streams/destroy.js:92:8)
        at emitErrorAndCloseNT (internal/streams/destroy.js:60:3) {
      statusCode: 0,
      responseText: 'Error: getaddrinfo EAI_AGAIN parity.goerli.ethnodes.brainbot.com\n' +
        '    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:66:26)'
    }
  }
}
2020-10-13T13:36:13.063Z [info] Closing server...
```

**Definition of Done**


**Steps to manually test the change (dApp)**

1. Try to trigger a network error on the CLI
2. Check the logged error is informative of which epic raised it
3. Check it's not `monitorUdcBalanceEpic`
